### DISCUSSION TITLE
Does consult-line-multi use orderless completion-style?

### DISCUSSION THREAD

**Discussion by Tokubara** [minad/consult#1066] [Q&A] [answered]
This is my `completion-style` setting:
```lisp
(setq completion-styles '(orderless baisc))
```
It works well with `consult-line`. But when I try to use `!` (negation) in `consult-line-multi`, it doesn't work. `!` is not interpreted as negation:
![image](https://github.com/user-attachments/assets/353b0c6e-7592-4c6f-a91e-30058f7f58c8)
 
How can I make negation and other meta-chars work in `consult-line-multi`?

**Comment by minad** [comment, marked as answer]
No, consult-line-multi doesn't use the completion style. It uses its own regexp compiler, which you could in principle replace with Orderless as described in the wiki. Note that even then negation will not work, since Orderless compiles negation to a predicate function by default (see `orderless-not`). But you can configure Orderless to use the less efficient `orderless-without-literal` instead for negation. All of this is a bit of a rabbit hole, I must admit.
